inspirehep/modules/workflows/tasks/manual_merging.py

```diff
@@ -110,16 +110,29 @@ def save_roots(obj, eng):
     """
     def _merge_roots(new_uuid, head_roots, update_roots):
         """Return the new roots to link to head."""
-        head_sources = [h.source for h in head_roots]
+        head_sources = {h.source: h for h in head_roots}
         for u in update_roots:
-            if u.source not in head_sources:
+            if u.source not in head_sources.keys():
```

Comment:
``u`` is a weird naming, probably something like ``updated_root`` or something similar